bolts/models/laftr.py

```diff
@@ -129,29 +133,27 @@ def _inference_step(self, batch: DataBatch, stage: str) -> Dict[str, Tensor]:
 
     def _loss_adv(self, s_pred: Tensor, batch: DataBatch) -> Tensor:
         # For Demographic Parity, for EqOpp is a different loss term.
-        s_target = batch.s.view(-1, 1)
-        y_target = batch.y.view(-1, 1)
         if self.fairness is FairnessType.DP:
-            s0 = self._adv_clf_loss(s_pred[batch.s == 0], s_target[batch.s == 0])
-            s1 = self._adv_clf_loss(s_pred[batch.s == 1], s_target[batch.s == 1])
+            s0 = self._adv_clf_loss(s_pred[batch.s == 0], batch.s[batch.s == 0])
+            s1 = self._adv_clf_loss(s_pred[batch.s == 1], batch.s[batch.s == 1])
             loss = (s0 + s1) / 2
         elif self.fairness is FairnessType.EO:
             loss = torch.tensor(0.0).to(self.device)
             for s, y in itertools.product([0, 1], repeat=2):
-                if len(batch.s[(batch.s == s) & (y_target == y)]) > 0:
+                if len(batch.s[(batch.s == s) & (batch.y == y)]) > 0:
                     loss += self._adv_clf_loss(
-                        s_pred[(s_target == s) & (y_target == y)],
-                        batch.s[(s_target == s) & (y_target == y)],
+                        s_pred[(batch.s == s) & (batch.y == y)],
+                        batch.s[(batch.s == s) & (batch.y == y)],
```

Comment:
for some reason I can't submit this as a suggestion
```python
                mask = (batch.s == s) & (batch.y == y)
                if len(batch.s[mask]) > 0:
                     loss += self._adv_clf_loss(
                         s_pred[mask], batch.s[mask]
```

Reply:
👍